### ISSUE TITLE
Missing execution logs for all executors if Test is triggered by TestSuite

### ISSUE THREAD

**Issue by tkonieczny** [kubeshop/testkube#3106]
**Describe the bug**
Execution logs aren't visible anymore if the test is triggered by TestSuite - both with CLI and Dashboard.

**To Reproduce**
1. Check one of the executor tests - for example `gradle-executor-smoke-jdk18` at Demo: https://demo.testkube.io/tests/executions/gradle-executor-smoke-jdk18/execution/63d272850d91f7f01d5c0407
2. `No logs` is visible in Dashboard, logs are also not visible in case of CLI:
```
kubectl testkube get execution 63d2767c3edbfc7e463223b9
client version is older than api version, please upgrade
ID:         63d2767c3edbfc7e463223b9
Name        executor-gradle-smoke-tests-gradle-executor-smoke-jdk17-19
Number:            19
Test name:         gradle-executor-smoke-jdk17
Type:              gradle:jdk17/test
Status:            passed
Start time:        2023-01-26 12:47:56.529 +0000 UTC
End time:          2023-01-26 12:48:46.521 +0000 UTC
Duration:          00:00:49
Labels:            core-tests=executors

  Variables:    1
  - TESTKUBE_GRADLE = true


Status Test execution completed with success 🥇
``` 
(this one is from `test` environment, but the result is the same at Demo)

Previous output (test executed 3 days ago):
```
kubectl testkube get execution 63ceaf90733a5d787edea863
client version is older than api version, please upgrade
ID:         63ceaf90733a5d787edea863
Name        executor-gradle-smoke-tests-gradle-executor-smoke-jdk17-1
Number:            1
Test name:         gradle-executor-smoke-jdk17
Type:              gradle:jdk17/test
Status:            passed
Start time:        2023-01-23 16:02:24.73 +0000 UTC
End time:          2023-01-23 16:02:46.91 +0000 UTC
Duration:          00:00:22
Labels:            core-tests=executors

  Variables:    1
  - TESTKUBE_GRADLE = true


Welcome to Gradle 7.4.2!

Here are the highlights of this release:
 - Aggregated test and JaCoCo reports
 - Marking additional test source directories as tests in IntelliJ
 - Support for Adoptium JDKs in Java toolchains

For more details see https://docs.gradle.org/7.4.2/release-notes.html

To honour the JVM settings for this build a single-use Daemon process will be forked. See https://docs.gradle.org/7.4.2/userguide/gradle_daemon.html#sec:disabling_the_daemon.
Daemon will be stopped at the end of the build 
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava
> Task :processTestResources NO-SOURCE
> Task :testClasses
> Task :test

BUILD SUCCESSFUL in 6s
2 actionable tasks: 2 executed

Status Test execution completed with success 🥇
```

**Expected behavior**
Execution logs should be displayed, execution stages should be visible.

**Version / Cluster**
```
Client Version 1.8.37
Server Version v1.8.37
Commit 8ddcc9c06162008ee8fd84b9284f431ac8de8206
Built by goreleaser
Build date 2023-01-25T23:14:27Z
```

**Comment by tkonieczny** [comment]
@vLia is it related to execution stages?

**Comment by vLia** [comment]
I did not look into the root-cause yet, but I am assuming it is related to the recent changes in adding logs to the output after the execution is finished. This is only occurring in tests ran in a test suite.

**Comment by tkonieczny** [comment]
So, I'm closing this one.

**Comment by vLia** [comment]
This was fixed in Testkube v1.9.8.

The problem was introduced by [this commit](https://github.com/kubeshop/testkube/commit/c26af07bdc437781fd01b59de8e18664ade5aac7), and was an issue caused by passing an object by value, and modifying a field that is pointer (pointer fields are copied as-is, therefore pointing to the original object).

The field containing the logs was overwritten when the tests ended and it was persisted for the first time. The suite also saved the results a second time, which now contained the empty object, overwriting the logs in the db. Fixed it by creating a deep copy and modifying that one during the first save.

A more elegant solution could've been refactoring these data structures, as they are already quite complicated, or refactoring the flow to not contain multiple db saves for the same thing. However, these solutions would've been more complex and error-prone.

**Comment by vsukhin** [comment]
thank you @vLia Good job! Might be @tkonieczny need to add a few more regression tests to check the logs in tests and testsuites
for @dejanzele for the future please check the changes for both cloud and OSS envs in order to avoid ithese errors
